containers/haskell/.devcontainer/Dockerfile

```diff
@@ -1,17 +1,35 @@
-# Use the [Choice] comment to indicate option arguments that should appear in VS Code UX
-#
-# [Choice] Haskell version: 9, 8
-ARG VARIANT=8
-FROM haskell:${VARIANT}
-
-# Use the [Option] comment to specify true/false arguments that should appear in VS Code UX
-#
+FROM debian:bullseye-slim
+
+ENV LANG C.UTF-8
+
+# common haskell + stack dependencies
+RUN apt-get update && \
+    apt-get install -y --no-install-recommends \
+        ca-certificates \
+        curl \
+        dpkg-dev \
+        git \
+        gcc \
+        gnupg \
+        g++ \
+        libc6-dev \
+        libffi-dev \
+        libgmp-dev \
+        libnuma-dev \
+        libtinfo-dev \
+        make \
+        netbase \
+        xz-utils \
+        zlib1g-dev && \
+    rm -rf /var/lib/apt/lists/*
```

Comment:
Are these needed here?
Most of the contents look like packages installed by `common-debian.sh`.
Please check to see which layer the other definitions are running `apt-get install` on.